compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -254,6 +254,46 @@ public boolean typeIncompatible(Location pos, BType actualType, BType expType) {
         return checkType(pos, actualType, expType, DiagnosticErrorCode.INCOMPATIBLE_TYPES) == symTable.semanticError;
     }
 
+    public BType getErrorTypes(BType bType) {
+        if (bType == null) {
+            return symTable.semanticError;
+        }
+
+        BType errorType = symTable.semanticError;
+
+        int tag = bType.tag;
+        if (tag == TypeTags.TYPEREFDESC) {
+            return getErrorTypes(Types.getReferredType(bType));
+        }
+        if (tag == TypeTags.ERROR) {
+            return bType;
+        }
+
+        if (tag == TypeTags.READONLY) {
+            return symTable.errorType;
+        }
+
+        if (tag != TypeTags.UNION) {
+            return errorType;
+        }
+
+        LinkedHashSet<BType> errTypes = new LinkedHashSet<>();
+        Set<BType> memTypes = ((BUnionType) bType).getMemberTypes();
+        for (BType memType : memTypes) {
+            BType memErrType = getErrorTypes(memType);
+
+            if (memErrType != symTable.semanticError) {
```

Comment:
We can use the type tag for this comparison.

Reply:
For types such as `symTable.semanticError`, `symTable.noType`, and builtin types like `symTable.errorType` and `symTable.intType` we expect only one instance of those types, so we can use reference equality.